soc/xtensa/intel_adsp/common/include/soc.h

```diff
@@ -112,4 +119,15 @@ extern bool soc_cpus_active[CONFIG_MP_NUM_CPUS];
  */
 int soc_adsp_halt_cpu(int id);
 
+static inline bool intel_adsp_ptr_executable(const void *p)
+{
+	return p >= (void *)_text_start && p <= (void *)_text_end;
+}
+
+static inline bool intel_adsp_ptr_is_sane(uint32_t sp)
```

Comment:
Not completely understanding the semantics here.  There is a lot of perfectly usable memory that this doesn't cover (non-heap regions, IMR, lpsram, etc..).   But then I don't know what this is needed for either, and "sane" isn't very specific.

Seems like this wants to be defined at a lower level, checking hp/lpsram and IMR memory as defined by the hardware kconfigs (and in both cached/uncached mappings).

Reply:
Yeah, those predicates come from ESP support, and I'm afraid that the mapping may not be straightforward to other platforms. From there, it seems to me that is just trying to avoid going into some "definitely not stack area" (like, text) while unwinding the stack.